src/transformers/integrations.py

```diff
@@ -1044,14 +1046,35 @@ def _initialize_run(self):
     @property
     def run(self):
         if self._run is None:
-            self._initialize_run()
+            self._reinitialize_run()
         return self._run
 
     def __del__(self):
         self._stop_run_if_exists()
 
+    @property
+    def _metadata_namespace(self):
+        return self.run[self._base_namespace_path]
+
+    def _log_integration_version(self):
+        self.run["source_code/integrations/transformers"] = version
+
+    def _log_trainer_parameters(self, args):
+        self._metadata_namespace["trainer_parameters"] = args.to_sanitized_dict()
+
+    def _log_model_parameters(self, model):
+        if model and hasattr(model, "config") and model.config is not None:
+            self._metadata_namespace["model_parameters"] = model.config.to_dict()
```

Comment:
I think we could store:
* "source_code/integrations/transformers"
* "trainer_parameters"
* "model_parameters"
Somewhere as constants.

Reply:
Done